scripts/extensions/sams/src/ui/SearchBar.tsx

```diff
@@ -1,25 +1,97 @@
 import * as React from 'react';
+import classNames from 'classnames';
 import {Icon} from 'superdesk-ui-framework/react';
+import nextId from 'react-id-generator';
 
 interface IProps {
+    type?: 'expanded' | 'collapsed' | 'boxed';
     placeholder: string; // defaults to light (white)
+    initialValue?: string;
+    focused?: boolean;
+    onChange?(newValue: string): void;
+    onSubmit(newValue: string): void;
 }
 
-export class SearchBar extends React.PureComponent<IProps> {
+interface IState {
+    value: string;
+}
+
+export class SearchBar extends React.Component<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            value: this.props.initialValue ?? '',
+        };
+
+        this.handleKeydown = this.handleKeydown.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.clearValue = this.clearValue.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    htmlId = nextId();
```

Comment:
I would have thought so too. Their docs state:
`NOTE: Don't initialize htmlId in React lifecycle methods like render(). htmlId should stay the same during component lifetime.` (https://www.npmjs.com/package/react-id-generator#nextid)

And their example shows placing this attribute in the class level (the ui-framework itself defines this variable in the same way).

Reply:
A constructor of a class isn't a React lifecycle method, so you can definitely put it there. The way it's defined in that library example is using syntax from ecmascript proposal for [class fields](https://github.com/tc39/proposal-class-fields). It's syntactic sugar and works for the most part the same as it was defined in the constructor.

I would like to avoid this syntax though in order to minimize the number of different patterns used across the codebase.

Reply:
This code is within the `ui` folder. It would be beneficial if we use the same patterns in this folder that the `UI-Framework` applies, so there is less changes they need to apply when porting this code across.